### PR TITLE
types(Partial): add partial reactions

### PR DESCRIPTION
**Pull request by DTrombett**
**Please describe the changes this PR makes and why it should be merged:**
This PR adds the type `AllowedPartials` for structures that can be partial. It also adds the support for partial reactions (the only partial structure without types) and changes how the `Partialize` type works (it should be only used by djs), so that it can easily extend partials structures.

Let me know if there are places where a reaction can be partial, other than in the event.

*Tests included!*

**Status and versioning classification:**
- This PR **only** includes non-code changes, like changes to documentation, README, etc.
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating

### CODE REVIEW

**Comment by DTrombett** [comment]
So these are new changes:
Moved the `partial` getter from DMChannel to Channel so that people can use `<Channel>.partial` even if the channel is not a dm (Shouldn't change a lot for non-typescript users).
Fixed `cleanContent` being not nullable in `Message` for docs.
Added all the necessary tests and fixed some properties being nullable in typings even if they're not when the Structure is not partial.

File: src/structures/Channel.js [hunk 1]
```diff
@@ -74,6 +74,15 @@ class Channel extends Base {
     return new Date(this.createdTimestamp);
   }
 
+  /**
+   * Whether this Channel is a partial
+   * @type {boolean}
+   * @readonly
+   */
+  get partial() {
+    return this instanceof DMChannel && typeof this.lastMessageId === 'undefined';
```
Comment (SpaceEEC) [inline src/structures/Channel.js:83]: This getter could just always return false and have dm channel (the only one that can be partial currently), do this check:
```suggestion
    return false;
```

File: src/structures/DMChannel.js [hunk 2]
```diff
@@ -50,15 +52,6 @@ class DMChannel extends Channel {
     this.lastPinTimestamp = data.last_pin_timestamp ? new Date(data.last_pin_timestamp).getTime() : null;
   }
 
-  /**
```
Comment (SpaceEEC) [inline src/structures/DMChannel.js:53]: This method should be restored.

**Review by SpaceEEC — CHANGES_REQUESTED** [review]

File: src/structures/Channel.js (hunk 1 above)
Comment (DTrombett) [reply src/structures/Channel.js:83]: Should I add in the jsdoc a note to say that this is always false if it's not a DMChannel?

**Review by vladfrangu — APPROVED** [review]

File: src/structures/Channel.js [hunk 3]
```diff
@@ -74,6 +74,16 @@ class Channel extends Base {
     return new Date(this.createdTimestamp);
   }
 
+  /**
+   * Whether this Channel is a partial
+   * Note that this is always false if the channel is not a DM
```
Comment (iCrawl) [inline src/structures/Channel.js:79]:
```suggestion
   * <info>This is always false outside of DM channels.</info>
```

Comment (DTrombett) [reply src/structures/Channel.js]: Added 👍

**Comment by DTrombett** [comment]
I'll fix last thing tomorrow.

**Comment by DTrombett** [comment]
Should be done! LMK if there's something to change.

**Review by SpaceEEC — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Comment by kyranet** [comment]
This needs a rebase, btw.

**Comment by DTrombett** [comment]
I'll do it tomorrow

**Comment by DTrombett** [comment]
Done!

**Review by iCrawl — APPROVED** [review]

**Pull request merged**